src/views/docs/en/guides/extend/plugins.md

```diff
@@ -154,20 +157,108 @@ rule-two
+
+#### Arguments
+
+All arguments arrive as a bag of options with the following properties:
+
+|Argument|Description|
+|---|---|
+|`arc`|Object representing the [parsed Architect project manifest](https://github.com/architect/parser) file for the current project|
+|`cloudformation`|The [CloudFormation JSON template][cfn-ref] making up the Architect project|
+|`stage`|The name of the environment; usually one of `testing`, `staging` or `production`; `testing` is provided when running in a [`sandbox`][sandbox] context whereas `staging` and `production` are provided at Architect CLI runtime when either `arc deploy staging` or `arc deploy production` are invoked, respectively|
+|`inventory`|An [Architect inventory object][inv] representing the current Architect project|
+
+#### Returns
+
+This method should always return an object. Each property on the object represents a variable name, and the value for each property contains the variable value.
+
+> 🏌️‍♀️ Protip: When this method is invoked in a pre-`deploy` context, acceptable values for the variables include CloudFormation JSON. This is essential to expose CloudFormation-managed infrastructure.
+
+#### Example `variables` Implementation
+
+The following example `variables` implementation demonstrates a plugin that creates a new S3 Bucket. It may be desirable to provide variables related to the location of and credentials for the bucket:
+
+```javascript
+module.exports = {
+  variables: function ({ arc, cloudformation, stage, inventory }) {
+    if (!arc['myS3Bucket']) return {} // if the user isn't using this plugin, return an empty object signifying no variables need exporting
+    const isLocal = stage === 'testing' // stage will equal 'testing' when running in sandbox, otherwise will be one of 'staging' or 'production'
+    const bucketName = `${arc.app}-newS3Bucket`
+    return {
+      bucketName,
+      accessKey: isLocal ? 'S3RVER' : { Ref: 'MyS3BucketCreds' },
+      secretKey: isLocal ? 'S3RVER' : { 'Fn::GetAtt': [ 'MyS3BucketCreds', 'SecretAccessKey' ] }
+    }
+  }
+}
+```
+
+Note that when running locally in [`sandbox`][sandbox], we provide some dummy set of credentials that the plugin hard-codes (and could check for when implementing the plugin [`sandbox.start`](#sandbox.start) method). Otherwise, when running in a pre-`deploy` context, we return CloudFormation JSON pointing to credentials the plugin added via the plugin's implemented [`package`](#package) method.
```

Comment:
I'm not following this

Reply:
I tried rewriting it, let me know what you think.